MultiSource/Benchmarks/Olden/treeadd/par-alloc.c

```diff
@@ -9,19 +9,18 @@
 #endif
 
 #include "tree.h"
-extern void *malloc(unsigned);
 
-tree_t *TreeAlloc (int level, int lo, int proc) {
+ptr<tree_t> TreeAlloc (int level, int lo, int proc) {
   if (level == 0)
     return NULL;
   else {
-    struct tree *new, *right, *left;
-    new = (struct tree *) malloc(sizeof(tree_t));
+    ptr<tree_t> new = NULL, right = NULL, left = NULL;
+    new = (ptr<tree_t>) malloc(sizeof(ptr<tree_t>));
```

Comment:
I'm sorry, I've been unclear. This should be `sizeof(tree_t)`, you're allocating space for the struct, not the ptr to the struct.